UPGRADING.md

```diff
@@ -1,6 +1,15 @@
 Upgrading Grape
 ===============
 
+### Upgrading to >= 0.18.1
+
+#### Aligned and clarified callback behavior of 405 handler and built-in OPTIONS handler
+
+When a request is made to a resource using an undefined method (HTTP 405) or a request is
+made to the built-in `OPTIONS` handler, only the `before` callbacks associated with the
+resource will be run.  Other callbacks (`before_validation`, `after_validation`, `after`)
```

Comment:
@jlfaber You're right. Indeed, the logging example should be placed in middleware stack however I'm not sure if it's worth breaking compatibility. 
Also if I'm newcomer, I'll wonder that both before and after behaviors are different.